packages/csvviewer/src/model.ts

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+
+import {
+  DataModel
+} from '@phosphor/datagrid';
+
+
+/**
+ * A data model implementation for in-memory DSV data.
+ */
+export
+class DSVModel extends DataModel {
+/**
+ * Create a data model with static CSV data.
+ *
+ * @param options - The options for initializing the data model.
+ */
+constructor(options: DSVModel.IOptions) {
+  super();
+  this._data = options.data;
+  this._delimiter = options.delimiter;
+  this._computeOffsets();
+}
+
+/**
+ * Get the row count for a region in the data model.
+ *
+ * @param region - The row region of interest.
+ *
+ * @returns - The row count for the region.
+ */
+rowCount(region: DataModel.RowRegion): number {
+  if (region === 'body') {
+    return this._offsets.length;
+  }
+  return 1;  // TODO multiple column-header rows?
+}
+
+/**
+ * Get the column count for a region in the data model.
+ *
+ * @param region - The column region of interest.
+ *
+ * @returns - The column count for the region.
+ */
+columnCount(region: DataModel.ColumnRegion): number {
+  // Parse first row, return that number of columns
+    if (region === 'body') {
+      if (this._offsets.length > 1) {
+        return this._data.slice(0, this._offsets[1]).split(this._delimiter).length;
```

Comment:
This should ideally be computed only once, as this method is called often.

Reply:
good point.